channeld/channeld.c

```diff
@@ -1034,11 +1034,30 @@ static struct bitcoin_signature *calc_commitsigs(const tal_t *ctx,
 	const u8 *msg;
 	struct bitcoin_signature *htlc_sigs;
 
+	/* Collect the htlcs for call to hsmd. */
+	struct simple_htlc **htlcs = tal_arr(tmpctx, struct simple_htlc *, 0);
+	size_t num_entries = tal_count(htlc_map);
+	for (size_t ndx = 0; ndx < num_entries; ++ndx) {
+		struct htlc const *hh = htlc_map[ndx];
+		if (hh) {
+			struct simple_htlc *simple =
+				new_simple_htlc(NULL,
```

Comment:
Please allocate using the `htlcs` array directly, since you're stealing right below it.

Reply:
fixed